ui/app/components/clients/history.js

```diff
@@ -92,7 +92,45 @@ export default class History extends Component {
   get latestUpgradeData() {
     // {id: '1.9.0', previousVersion: null, timestampInstalled: '2021-11-03T10:23:16Z'}
     // version id is 1.9.0 or earliest upgrade post 1.9.0, timestamp is RFC3339
-    return this.args.model.versionHistory[0] || null;
+    // need to return the last in the array as the data is in ascending order
+    return this.args.model.versionHistory.at(-1) || null;
+  }
+
+  get allMinorVersionsInRange() {
+    let newArray = [];
+    // ARG TODO testing unsure if the getActivityResponse is the correct one here.
+    let startTime = new Date(this.getActivityResponse.startTime);
+    let endTime = new Date(this.getActivityResponse.endTime);
+    this.args.model.versionHistory.forEach((object) => {
+      let versionDate = new Date(object.timestampInstalled);
+      // compare against the API response to see if date is within the range of returned response.
+      if (isAfter(versionDate, startTime) && isBefore(versionDate, endTime)) {
+        // push the minor version number e.g. 9
+        let upgradeVersionArray = object.id.split('.');
+        newArray.push(upgradeVersionArray[1]);
+      }
+    });
+    return newArray; // return [9,9,10] shows all the minor numbers that fall within the range of the returned API response.
+  }
+
+  get versionUpdateText() {
+    // count how many times a minor number occurs inside the array of minors within the API range.
```

Comment:
Do we need to know the count here? I'm wondering if we could simplify this a by using `.includes` instead.
```
let includesVersion9 = this.allMinorVersionsInRange.includes(9)
let includesVersion10 = this.allMinorVersionsInRange.includes(10)

if (includesVersion 9 && includesVersion10) {
...text for both }
if (includesVersion9) {
...text for 1.9
}
if (includes Version10) {
..text for 1.10 
}